pkg/resources/scim_integration.go

```diff
@@ -71,37 +75,72 @@ var scimIntegrationSchema = map[string]*schema.Schema{
 	},
 }
 
-// SCIMIntegration returns a pointer to the resource representing a network policy.
 func SCIMIntegration() *schema.Resource {
 	return &schema.Resource{
 		CreateContext: CreateContextSCIMIntegration,
 		ReadContext:   ReadContextSCIMIntegration,
 		UpdateContext: UpdateContextSCIMIntegration,
 		DeleteContext: DeleteContextSCIMIntegration,
-
+		CustomizeDiff: customdiff.All(
+			BoolComputedIf("sync_password", "SYNC_PASSWORD", func(client *sdk.Client, id sdk.AccountObjectIdentifier) ([]Property, error) {
+				props, err := client.SecurityIntegrations.Describe(context.Background(), id)
+				res := make([]Property, len(props))
+				for i := range props {
+					res[i] = props[i]
+				}
+				return res, err
+			}),
+		),
 		Schema: scimIntegrationSchema,
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
+		SchemaVersion: 1,
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Version: 0,
+				// setting type to cty.EmptyObject is a bit hacky here but following https://developer.hashicorp.com/terraform/plugin/framework/migrating/resources/state-upgrade#sdkv2-1 would require lots of repetitive code; this should work with cty.EmptyObject
+				Type:    cty.EmptyObject,
+				Upgrade: v091ScimIntegrationStateUpgrader,
+			},
+		},
+	}
+}
+
+func v091ScimIntegrationStateUpgrader(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
```

Comment:
let's move it to a separate file like with other upgraders

Reply:
Done.